src/Arcus.Messaging.ServiceBus.Core/ServiceBusMessageBuilder.cs

```diff
@@ -57,7 +57,23 @@ public static ServiceBusMessageBuilder CreateForBody(object messageBody, Encodin
         public ServiceBusMessageBuilder WithOperationId(string operationId)
         {
             Guard.NotNullOrWhitespace(operationId, nameof(operationId), "Requires a non-blank operation ID for the Azure Service Bus message");
-            _operationId = operationId;
+
+            _operationIdProperty = new KeyValuePair<string, object>(null, operationId);
+            return this;
+        }
+
+        /// <summary>
+        /// Adds an <paramref name="operationId"/> as an application property in the <see cref="ServiceBusMessage.ApplicationProperties"/>.
+        /// </summary>
+        /// <param name="operationId">The unique identifier for this operation.</param>
+        /// <param name="operationIdPropertyName">The name of the application property where the <paramref name="operationId"/> will be set.</param>
```

Comment:
```suggestion
        /// <param name="operationIdPropertyName">The custom name that must be given to the application property that contains the <paramref name="operationId"/>.</param>
```

Reply:
Ok, great, maybe we should have the same format for the other ones as well. I'll update this. 👍 thx!